### ISSUE TITLE
Not working in IE9 and below

### ISSUE THREAD

**Issue by rusticblonde** [jonathantneal/flexibility#87]
Hi,

Ive tried to get this working in IE9 and no luck whatsoever.

Ive placed the prefixes in, and no luck. Can someone please explain what im doing wrong.

http://codepen.io/rusticblonde/pen/xgJRdj

Thanks
Kirsty

**Comment by luisbajana** [comment]
Having the same problems, it's not working...

**Comment by moesphemie** [comment]
Same here with 2.0.1

**Comment by luisbajana** [comment]
I solved it using the 1.x version of this plugin, apparently the problem affects only 2.x versions.

**Comment by moesphemie** [comment]
Do you have to do any more than add `-js-display: flex;` to the CSS file?

**Comment by luisbajana** [comment]
Yes you have to initialize it in your HTML (this is only gonna be execute it in IE)


I added prefixes for old webkit based browsers like safari in iPad 2 for example, I used this routine in gulp to do two things, add the prefixes for old webkit based browsers (autoprefixer()), and the postcss-flexibility to make this plugin work in IE. 

gulp.task('autoprefixer', function () {
    var postcss      = require('gulp-postcss');
    var sourcemaps   = require('gulp-sourcemaps');
    var autoprefixer = require('autoprefixer');
    var path = require('path');

    return gulp.src('./*.css')
        .pipe(sourcemaps.init())
        .pipe( postcss([ require('postcss-flexibility') ]) )
        .pipe(postcss([ autoprefixer() ]))
        .pipe(sourcemaps.write('.'))
        .pipe(gulp.dest('./css'));
});

**Comment by moesphemie** [comment]
What do you mean by initialize? 
In my default.pug file i'm loading it like this:

`script(src='/bower_components/flexibility/dist/flexibility.js')`

I don't have to execute it in my main.js or something?

**Comment by luisbajana** [comment]
You're just loading the plugin, you have to call the function passing an object as a parameter, in this case I'm passing document.body.

flexibility(document.body);

**Comment by moesphemie** [comment]
Ah thank you. When i follow your instruction i get this error in IE9:

SCRIPT5002: Function expected 
main.js, line 159 character 1
SCRIPT5007: Unable to get value of the property '-js-display': object is null or undefined 
flexibility.js, line 1 character 11784

**Comment by katiasmet** [comment]
I have the same error. @moesphemie How did you fix this?

**Comment by moesphemie** [comment]
@katiasmet unfortunately i couldn't get this to work. Instead i used the good old floats in combination with matchHeight in order to get the same height for all my divs.

**Comment by yale8848** [comment]
lib 2.0.1

```
<div class="box box0">
<span class="item"></span>
</div>
```
```
.box {
    display: flex;
    width: 50px;
    height: 50px;
    background-color: #fff;
    border-radius: 10px;
    margin-left: 50px;
    margin-top: 50px;
    padding: 2px;
}
.box .item {
    display: inline-block;
    width: 15px;
    height: 15px;
    background-color: #000;
    border-radius: 50%
}
.wrapper .box0 {
    justify-content: center;
}
```
right display:

![right](https://cloud.githubusercontent.com/assets/2555005/25212741/d1b5a646-25be-11e7-8d04-6340748ecdca.png)

but in IE9 display:

![error](https://cloud.githubusercontent.com/assets/2555005/25212752/e5a41c78-25be-11e7-8ff6-6520b207d743.png)

```
<script type='text/javascript' src="./js/flexibility.js">
</script>
<script type='text/javascript'>

window.onload=function(){
flexibility(document.body);
}

</script>
```

```
.wrapper .box0 {
    justify-content: center;
    -js-display: flex;
}
```

> then i change lib to 1.0.6 ,an it works well in IE9

**Comment by chrispepper1989** [comment]
Thanks for this info yale8848, I feel a bit closer than before, but now i'm getting this basic JS error from the flexibility.js:

SCRIPT5009: 'module' is undefined
flexibility.js (1,502)

as its minified it's hard to see what it's complaining about specifically....

**Comment by squadjot** [comment]
https://jonneal.dev/flexibility/ 
I tested this in ie8, and it does not work

If you're having trouble opening a "real" ie8 browser put this in a vbs file and run it.
```
Set objShell = WScript.CreateObject("WScript.Shell")
Set objIE = WScript.CreateObject("InternetExplorer.Application", "IE_")
objIE.Visible = 1
objShell.AppActivate objIE
objIE.Navigate2 "about:blank"
```
